integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/generator.py

```diff
@@ -65,13 +65,13 @@ class AmazonBedrockGenerator:
     """
 
     SUPPORTED_MODEL_PATTERNS: ClassVar[Dict[str, Type[BedrockModelAdapter]]] = {
-        r"amazon.titan-text.*": AmazonTitanAdapter,
-        r"ai21.j2.*": AI21LabsJurassic2Adapter,
-        r"cohere.command-[^r].*": CohereCommandAdapter,
-        r"cohere.command-r.*": CohereCommandRAdapter,
-        r"(.+\.)?anthropic.claude.*": AnthropicClaudeAdapter,
-        r"meta.llama.*": MetaLlamaAdapter,
-        r"mistral.*": MistralAdapter,
+        r"(eu\.|us\.)?amazon.titan-text.*": AmazonTitanAdapter,
```

Comment:
Let's use the reg-ex seen [here](https://github.com/deepset-ai/haystack-core-integrations/pull/1120) - This will also handle other regions that might get supported in the future.

If we know that AWS uses the [ISO country code](https://en.wikipedia.org/wiki/List_of_ISO_3166_country_codes), we should also restrict the length.

Reply:
Updated the regex to support ISO Alpha-2 country code prefixes like so: `([a-z]{2}\.)?`.